server/graphql/v1/mutations/expenses.js

```diff
@@ -703,67 +704,91 @@ export async function payExpense(req, args) {
       );
     }
 
-    // If the amount of the expense is larger than i.e. $1000 enforce 2FA if the user has it turned on
-    const is2FARequiredForPayoutMethod = [PayoutMethodTypes.PAYPAL, PayoutMethodTypes.BANK_ACCOUNT].includes(
-      payoutMethodType,
+    // 2FA for payouts
+    const isTwoFactorAuthenticationRequiredForPayoutMethod = [
+      PayoutMethodTypes.PAYPAL,
+      PayoutMethodTypes.BANK_ACCOUNT,
+    ].includes(payoutMethodType);
+    const hostHasPayoutTwoFactorAuthenticationEnabled = get(host, 'settings.payoutsTwoFactorAuth.enabled', false);
+    const hostPayoutTwoFactorAuthenticationRollingLimit = get(
+      host,
+      'settings.payoutsTwoFactorAuth.rollingLimit',
+      1000000,
     );
-    const hostHasPayout2FAEnabled = get(host, 'settings.payoutsTwoFactorAuth.enabled', false);
-    const hostPayout2FAExpenseAmount = get(host, 'settings.payoutsTwoFactorAuth.expenseAmount', 100000);
-    if (
-      is2FARequiredForPayoutMethod &&
+    const useTwoFactorAuthentication =
+      isTwoFactorAuthenticationRequiredForPayoutMethod &&
       !args.forceManual &&
-      hostHasPayout2FAEnabled &&
-      expense.amount >= hostPayout2FAExpenseAmount
-    ) {
-      enforceTwoFactorAuthenticationOnPayouts(req, args.twoFactorAuthenticatorCode);
+      hostHasPayoutTwoFactorAuthenticationEnabled;
+
+    if (useTwoFactorAuthentication) {
+      await rollingPayoutLimitTwoFactorAuthentication(
```

Comment:
Not a blocker but I have a hard time understanding the purpose of this method by looking at its name. I think a verb is missing.

Maybe `handleTwoFactorAuthenticationPayoutLimit`. Or `twoFactorAuthentication.handlePayoutLimit` when using another import/export style.